### PR TITLE
Replace flake8 with ruff

### PR DESCRIPTION
**Pull request by aljazerzen**
Replaces flake8 with ruff. Also fixes a few lint errors that show up because of that.

Total time on my machine:
- flake 1.415 sec
- ruff 0.05 sec

The important question: what have I done with the `from typing import *` custom extension that we had? I've disabled the rule. I plan to change all `from typing import *` into `from typing import List, Dict` before merging this PR.

### CODE REVIEW

**Comment by aljazerzen** [comment]
Re `import *`: A pretty good way of doing this would be search and replace with all names in `typing`:
```
from typing import Any, ClassVar, Generic, Optional, Tuple, Type, TypeVar, Union, AbstractSet, Iterable, Iterator, MutableMapping, MutableSequence, MutableSet, Sequence, Dict, List, Set, cast, overload
```
... and then running `ruff check --fix .`, which would remove unused imports.

**Comment by aljazerzen** [comment]
https://github.com/edgedb/edgedb/pull/6874 has a few problems with the new f-string syntax and flake8, upgrading to ruff fixes that.

**Comment by msullivan** [comment]
Should we investigate whether ruff would take a patch to support `from typing import *` nicely?

**Review by msullivan — APPROVED** [review]
I guess. I really do hate the typing stuff though.

File: pyproject.toml [hunk 1]
```diff
@@ -213,3 +210,19 @@ ignore_errors = true
...
+    "B008",
+    "B904",
+    "B905",
+    "B019",
+    "E731",
+    "E741",
+]
+line-length = 80
+indent-width = 4
+target-version = "py310"
```
Comment (elprans) [inline pyproject.toml:227]: No need for this, because `ruff` looks at `requires-python` that we already have above

**Review by elprans — APPROVED** [review]

**Pull request merged**